iceoryx_utils/include/iceoryx_utils/internal/cxx/storable_function.inl

```diff
@@ -0,0 +1,250 @@
+{
+    m_vtable.copy(other, *this);
+}
+
+template <typename S, typename ReturnType, typename... Args>
+storable_function<S, signature<ReturnType, Args...>>::storable_function(storable_function&& other) noexcept
+    : m_vtable(other.m_vtable)
+{
+    m_vtable.move(other, *this);
+}
+
+template <typename S, typename ReturnType, typename... Args>
+storable_function<S, signature<ReturnType, Args...>>&
+storable_function<S, signature<ReturnType, Args...>>::operator=(const storable_function& rhs) noexcept
+{
+    if (&rhs != this)
+    {
+        // note: src vtable is needed for destroy, then changed to src vtable
+        m_vtable.destroy(*this);
+        m_function = nullptr; // only needed when the src has no object
+        m_vtable = rhs.m_vtable;
+        m_vtable.copy(rhs, *this);
+    }
+
+    return *this;
+}
+
+template <typename S, typename ReturnType, typename... Args>
+storable_function<S, signature<ReturnType, Args...>>&
+storable_function<S, signature<ReturnType, Args...>>::operator=(storable_function&& rhs) noexcept
+{
+    if (&rhs != this)
+    {
+        // note: dest (this) vtable is needed for destroy, then changed to source (rhs) vtable
+        m_vtable.destroy(*this);
+        m_function = nullptr; // only needed when the source has no object
+        m_vtable = rhs.m_vtable;
+        m_vtable.move(rhs, *this);
+    }
+
+    return *this;
+}
+
+template <typename S, typename ReturnType, typename... Args>
+storable_function<S, signature<ReturnType, Args...>>::~storable_function() noexcept
+{
+    m_vtable.destroy(*this);
+}
+
+
+template <typename S, typename ReturnType, typename... Args>
+ReturnType storable_function<S, signature<ReturnType, Args...>>::operator()(Args... args)
+{
+    auto r = m_function(std::forward<Args>(args)...);
+    return r;
+}
+
+template <typename S, typename ReturnType, typename... Args>
+storable_function<S, signature<ReturnType, Args...>>::operator bool() noexcept
+{
+    return m_function.operator bool();
+}
+
+template <typename S, typename ReturnType, typename... Args>
+void storable_function<S, signature<ReturnType, Args...>>::swap(storable_function& f) noexcept
+{
+    storable_function tmp = std::move(f);
+    f = std::move(*this);
+    *this = std::move(tmp);
+}
+
+template <typename S, typename ReturnType, typename... Args>
+void storable_function<S, signature<ReturnType, Args...>>::swap(storable_function& f, storable_function& g) noexcept
+{
+    storable_function tmp = std::move(f);
+    f = std::move(g);
+    g = std::move(tmp);
+}
+
+template <typename S, typename ReturnType, typename... Args>
+template <typename Functor, typename>
+void storable_function<S, signature<ReturnType, Args...>>::storeFunctor(const Functor& functor) noexcept
+{
+    ///@todo we may need something more to get the correct type in general...
+    using StoredType = typename std::remove_reference<Functor>::type;
+    auto p = m_storage.template allocate<StoredType>();
+
+    if (p)
```

Comment:
This condition check could be more self-documenting.

Reply:
Hm, we should not go overboard with this. It is quite low level. When the allocation fails we get a nullptr, as usual in many APIs. consider writing something like 

```
const auto allocationSucceeded = p != nullptr;
if(allocationSucceeded) {
...
}
```
overkill.

Reply:
Yeah, my argument would be naming `p` as `ptr` instead makes it immediately clear we are checking a nullptr.
I won't argue for it further though. As you like.